subsys/bluetooth/host/scan.c

```diff
@@ -37,6 +41,14 @@ static sys_slist_t pa_sync_cbs = SYS_SLIST_STATIC_INIT(&pa_sync_cbs);
 void bt_scan_reset(void)
 {
 	scan_dev_found_cb = NULL;
+#if defined(CONFIG_BT_EXT_ADV)
+	if (ext_scan_buf) {
+		net_buf_reset(ext_scan_buf);
```

Comment:
The `ext_scan_buf` is currently reset here in `bt_scan_reset` and when an advertising report is completed. 

I wonder if it should be reset once scan is started; could we end in the situation where

1. Host receives first fragment and stores it in `ext_scan_buf`
2. Host stops scan (by request of upper layers)
3. Host starts scan again
?

If yes, I think we need to reset the buffer in scan start. If no, can it happen when we reset?

Reply:
I agree with your observation, we need to reset it when starting scanning. There is no need to reset it on HCI reset.

Reply:
- [ ] Reset buffer on scan start